erpnext/accounts/doctype/payment_request/payment_request.py

```diff
@@ -376,8 +376,8 @@ def make_payment_request(**args):
 		if args.order_type == "Shopping Cart" or args.mute_email:
 			pr.flags.mute_email = True
 
-		pr.insert(ignore_permissions=True)
 		if args.submit_doc:
+			pr.insert(ignore_permissions=True)
```

Comment:
Why? This is a huge breaking change. Earlier it was inserting a document and now it just returns in-memory document. 

This breaks:
1. API. 
2. other's code, in this case even ERPNExt: https://github.com/frappe/erpnext/blob/b58b1343bb80f3442f49728e965f834a35d2f257/erpnext/education/doctype/fees/fees.py#L78-L80